oak_channel/src/frame.rs

```diff
@@ -117,9 +117,11 @@ impl Framed {
             let body_length: usize = length
                 .checked_sub(BODY_OFFSET)
                 .expect("body length underflow");
-            let body = allocate_fn(body_length);
-            self.inner.read(body)?;
-            body
+            let tail = message_buffer.len();
```

Comment:
It looks like https://doc.rust-lang.org/std/io/trait.Read.html#method.read_buf would do exactly what we need, but I suspect we cannot use it because it relies on std::io::Read, is that correct?

Reply:
BTW, "This is a nightly-only experimental API." - is this of any concern? I noticed that I could compile Oak with the nightly toolchain only, IIRC due to some #feature annotation.

Reply:
nightly-only is ok, we are using a nightly compiler anyways (though it may need to be updated). but it may still not work if it relies on std, since this code must still work in no_std mode, right?

Reply:
yup, we need no_std support for the code running inside the enclave. Atm we use our own read trait (that defaults to std if available). Atm our no_std implementation has equivalent for [read_buf](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_buf). Note that it may be possible to write no_std compatible implementation of it however, in which case we could go down that route.

Reply:
I see. So references to std::* are not allowed, even though they would provide means for a more efficient implementation (std::io::Read::read_buf()). In that case we can resolve the thread, but should we make a note to add read_buf to our own no_std impl?

Reply:
I think we can close this thread. Not sure if making a note is worth it tbh, adding that implementation would be non trivial.

Reply:
Sounds good.